src/test/java/jenkins/branch/harness/BranchProjectFactoryImpl.java

```diff
@@ -42,7 +42,7 @@ public BranchProjectFactoryImpl() {}
 
     @Override
     public FreeStyleProject newInstance(Branch branch) {
-        FreeStyleProject job = new FreeStyleProject(getOwner(), branch.getName());
+        FreeStyleProject job = new FreeStyleProject(getOwner(), branch.getEncodedName());
```

Comment:
perhaps we should be setting the display name to `branch.getName()` if we are switching to encoded name